conn_test.go

```diff
@@ -11,17 +11,44 @@ import (
 	"sync/atomic"
 	"time"
 
+	"github.com/libp2p/go-libp2p-core/control"
 	ic "github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/libp2p/go-libp2p-core/network"
 	"github.com/libp2p/go-libp2p-core/peer"
 	tpt "github.com/libp2p/go-libp2p-core/transport"
-	filter "github.com/libp2p/go-maddr-filter"
+
 	quicproxy "github.com/lucas-clemente/quic-go/integrationtests/tools/proxy"
 	ma "github.com/multiformats/go-multiaddr"
+	manet "github.com/multiformats/go-multiaddr-net"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 )
 
+type localhostMockGater struct {
```

Comment:
@raulk I know this will make you cringe and add some entropy to thy life. But, I have added a TODO at https://github.com/libp2p/go-libp2p/issues/930 to replace this with the "Functional Gater" that we lan to implement. 

For all that you hold dear on this good Earth, rest assured, this shall be fixed.